module/VuFind/src/VuFind/Search/Factory/AbstractSolrBackendFactory.php

```diff
@@ -482,14 +482,14 @@ protected function getHideFacetValueListener(
         BackendInterface $backend,
         Config $facet
     ) {
-        if (!isset($facet->HideFacetValue)
-            || ($facet->HideFacetValue->count()) == 0
-        ) {
-            return null;
-        }
+        $hideFacetValue = (isset($facet->HideFacetValue))
+            ? $facet->HideFacetValue->toArray() : [];
+        $hideAllFacetValuesExcept = (isset($facet->HideAllFacetValuesExcept))
+            ? $facet->HideAllFacetValuesExcept->toArray() : [];
```

Comment:
We don't want to create the listener if it's not going to do any work, so I think there should be a check here that says:

```
if (empty($hideFacetValue) && empty($hideAllFacetValuesExcept)) {
    return null;
}
```

Do you have time to add and test that? Thanks!

Reply:
tested and added

Reply:
Thank you!